arwen/host/execution.go

```diff
@@ -274,6 +358,10 @@ func (host *vmHost) execute(input *vmcommon.ContractCallInput) error {
 		return arwen.ErrInitFuncCalledInRun
 	}
 
+	if host.isUpgradeFunctionBeingCalled() {
```

Comment:
This should be never reached, why is this protection here?

Reply:
Removed constraint.

In order to avoid conflicts with work in progress regarding indirect execution, SC upgrade initiated by parent SC will be done in a future PR. [EN-6058](
https://elrondnetwork.atlassian.net/browse/EN-6058).